ast/src/analyzed/serialized.rs

```diff
@@ -0,0 +1,84 @@
+use std::path::PathBuf;
+
+use powdr_number::{FieldElement, KnownField};
+use serde::{Deserialize, Serialize};
+
+use super::Analyzed;
+
+const MAGIC: u32 = 0x4c_41_50_42;
```

Comment:
This MAGIC number might be useless in the end, as the deserialization will fail anyway if the file is not in the good format.

Reply:
But if Powdr becomes so important that file explorer wants to display something about the file, it will help ;).

Reply:
how did you pick the bytes

Reply:
Randomly, and checked if they existed in a repository of magic files. 
Any other series of bytes will do if you have a specific idea :)

Reply:
For futur references, the UNIX style magic format for this particular file can be defined as such:
```
8       belong          0x4c415042      Powdr PIL binary object
```

Reply:
we could use 0x706f776472 for `powdr` :)

Reply:
wait that's too many bytes

Reply:
yea the original sequence is fine!

Reply:
> we could use 0x706f776472 for `powdr` :)

Haha that was my first idea also, we can definitely use it tho, because we can define magic files based on strings. Your call :p

Reply:
Ok I got it working, pushing right now